File: akka-essentials/Chapter9/AkkaRemotingExample/RemoteNodeApp/src/main/scala/org/akka/essentials/remotenode/RemoteActor.scala
package org.akka.essentials.remotenode

import akka.actor.{ActorRef, Actor}

class RemoteActor extends Actor {
    def receive: Receive = {
        case message: String =>
            // Get reference to the message sender and reply back
            sender.tell(message + " got something", ActorRef.noSender)
    }
}

File: akka-essentials/Chapter9/AkkaRemotingExample/RemoteNodeApp/src/main/java/org/akka/essentials/remotenode/java/RemoteActor.java
package org.akka.essentials.remotenode.java;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;

public class RemoteActor extends UntypedActor {
	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof String) {
			// Get reference to the message sender and reply back
			getSender().tell(message + " got something", ActorRef.noSender());
		}
	}
}
